### ISSUE TITLE
async loading model 

### ISSUE THREAD

**Issue by enricooliva** [formly-js/ngx-formly#1255]
```
[ ] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ x] support request
```

**Current behavior**       
```
 ngOnInit() {    
    this.route.params.subscribe(params => {
        this.service.getConvenzioneById(params['id']).subscribe((data) => {
          this.model = data;
        });
    });
}
```
When the model is populated for the first time all the **field.formControl.valueChanges** emit the event.

How to avoid to listen the first event?

**Comment by aitboudad** [comment]
the easy way is to avoid init the form until model is ready, using 
```html
<formly-form *ngIf`
```

**Comment by enricooliva** [comment]
I used:
` <formly-form *ngIf="model"`
and in my template
`       
lifecycle: {          
          onInit: (form, field) => {                                  
            field.formControl.valueChanges.pipe(
              distinctUntilChanged(),             
              takeUntil(this.onDestroy$),              
              tap(selectedField => {               
 `

* onInit is called only with field has value 
* valueChanges is called only after a user modification
Seems working well.

Drawback is a completely empty page until it arrives the model
I'm not satisfy 100% seems the form + model need to be provided at the same time.

Other stuff related to the init of the form:  I need in the template access to other part of the model is possible? 

 ```
onInit: (form, field) => {         
//HERE ... other part of the model ...                         
            field.formControl.valueChanges.pipe(
```

**Comment by enricooliva** [comment]
@aitboudad I have problem in reloading form.

**Comment by aitboudad** [comment]
I'm going to close since it's not clear, please elaborate to get an answer.

**Comment by enricooliva** [comment]
@aitboudad you are right. Thank you for your response.  

I have this button with this disabled condition: `!form.valid || !form.dirty` 
` <button class="btn btn-outline-primary border-0 rounded-0" [disabled]="!form.valid || !form.dirty" (click)="onSubmit()" >   `
`<form *ngIf='model' [formGroup]="form" >`

```
onReload (){
      this.model = null; 
      let id = this.model.id;
      this.service.getConvenzioneById(id).subscribe((data) => {                  
        this.model = data;            
      });
}
```

I call the onReload function.
The form remain in dirty state.

If I did `this.form = new FormGroup({});`  the form became not modified but it is disconnected from the button.

**Comment by aitboudad** [comment]
try with `this.form.reset()`

**Comment by enricooliva** [comment]
Hi @aitboudad the result of the form now is coerent. Thank you.

But the side effect is the form disappear during the update process. Any suggestion?
```
   this.model = null;
   this.service.updateConvenzione(tosubmit, tosubmit.id).subscribe(
        result => {
          this.form.reset();          
          this.isLoading = false;          
          this.id = result.id;
          this.model = result;                    
        },
```
![step1](https://user-images.githubusercontent.com/11843561/47713341-5c53a680-dc3a-11e8-8b71-e22b1b3f01b4.PNG)
![step2](https://user-images.githubusercontent.com/11843561/47713342-5cec3d00-dc3a-11e8-954a-a0482973bfc4.PNG)
![step3](https://user-images.githubusercontent.com/11843561/47713343-5cec3d00-dc3a-11e8-8017-9dbed7e9c178.PNG)

**Comment by aitboudad** [comment]
ok instead of setting model to `null` rely on a locale variable (for example `isLoading`) to ignore emitted value during `updateConvenzione` and use `options.reset()` to update model value:

```ts
// field config
field.formControl.valueChanges.pipe(
  distinctUntilChanged(),
  takeUntil(this.onDestroy$), 
  filter(() => !this.isLoading),
  ...
)

// updateConvenzione
   this.service.updateConvenzione(tosubmit, tosubmit.id).subscribe(
        result => {
          this.id = result.id;
          this.options.resetModel(result);          
          this.isLoading = false;              
        },
  )
```

**Comment by enricooliva** [comment]
Ok great! 
I have a valueChanges inside a formly template, how to pass the local variable `isLoading` inside the template or set it externally?

**Comment by aitboudad** [comment]
pass it through `formState`
https://formly-js.github.io/ngx-formly/examples/form-options/form-state

**Comment by enricooliva** [comment]
I like the solution with `form-state`.

I lost only one behavior in select 

```
          {
            key: 'tipoemittenti_codice',
            type: 'select',
            className: "col-md-6",
            templateOptions: {
              options: [],
              valueProp: 'codice',
              labelProp: 'descrizione',
              label: 'Autorizzato da',
              required: true
            },
            lifecycle: {
              onInit: (form, field) => {
                field.templateOptions.options = [this.model.tipoemittente];
                this.service.getEmittenti().subscribe((data)=> {
                  field.templateOptions.options = data
                });
              }
            }
```

Using the `*ngIf` on model when start the onInit the model is populated and 
the set ` field.templateOptions.options = [this.model.tipoemittente];` in the onInit 
work well.

@aitboudad is better use` lifecycle onChange `or register` form.get('dipartimemto_cd_dip').valueChanges.pipe(` and control `isLoading`

**Comment by rdev-software** [comment]
Any update ? 

When i try to load model with 
`<ion-content *ngIf="loadedGame | async as game" padding>`

Validation does not work

**Comment by aitboudad** [comment]
@rdev-software not sure what you mean, please elaborate in a new issue instead.

**Comment by rdev-software** [comment]
@aitboudad That if I have observable **loadedGame** and I want to async load it in parent in this case ion-content and place dynamic form inside and use loaded object **game** inside, validation does not work (and maybe other things as well as per this thread)

**Comment by aitboudad** [comment]
@rdev-software please replicate what you've done using our starter example https://stackblitz.com/edit/ngx-formly-ui-ionic in order to see what going wrong + fill a new issue.
